integrations/nvidia/src/haystack_integrations/components/embedders/nvidia/_nim_backend.py

```diff
@@ -50,3 +50,20 @@ def embed(self, texts: List[str]) -> Tuple[List[List[float]], Dict[str, Any]]:
         embeddings = [e["embedding"] for e in sorted(data["data"], key=lambda e: e["index"])]
 
         return embeddings, {"usage": data["usage"]}
+
+    def models(self) -> List[Model]:
+        url = f"{self.api_url}/models"
+
+        res = self.session.get(
+            url,
+            timeout=REQUEST_TIMEOUT,
+        )
+        res.raise_for_status()
+
+        data = res.json()["data"]
+        models = []
+        for element in data:
+            assert "id" in element, f"No id found in {element}"
```

Comment:
If this is a user-facing error that can happen, we should raise an exception instead (and use a more descriptive error message).